### PR TITLE
Add custom JSDoc template, add @signal tag 

### PR DESCRIPTION
**Pull request by ingerjm0**
This PR:
* Adds a custom JSDoc template for High Fidelity's API documentation (based off of DocDash)
* Configures JSDoc to use custom template
* Adds @signal tag to separate signals from methods

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17559/HighFidelity-Beta-PR15009-75000d9.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17559/HighFidelity-Beta-Interface-PR15009-75000d9.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17559/HighFidelity-Beta-PR15009-75000d9.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17559/HighFidelity-Beta-Interface-PR15009-75000d9.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17559/HighFidelity-Beta-PR15009-75000d9.apk

File: tools/jsdoc/plugins/hifi.js [hunk 1]
```diff
@@ -107,25 +107,38 @@ exports.handlers = {
...
 };
 
+// Functions for adding @signal custom tag
+/** @private */
+function setDocletKindToTitle(doclet, tag) {
+    doclet.addTag( 'kind', tag.title );
+}
+
+function setDocletNameToValue(doclet, tag) {
+    if (tag.value && tag.value.description) { // as in a long tag
```
Comment (ctrlaltdavid) [inline tools/jsdoc/plugins/hifi.js]: There should be a space between `if` and its `(`.

File: tools/jsdoc/plugins/hifi.js [hunk 2]
```diff
@@ -107,25 +107,38 @@ exports.handlers = {
...
+        doclet.addTag('name', tag.value.description);
+    }
+    else if (tag.text) { // or a short tag
```
Comment (ctrlaltdavid) [inline tools/jsdoc/plugins/hifi.js]: `else if` shouldn't start on a new line - the line with it should start, `} else if`

Comment (ingerjm0) [reply tools/jsdoc/plugins/hifi.js]: Fixed!

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17575/HighFidelity-Beta-PR15009-761a922.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17575/HighFidelity-Beta-Interface-PR15009-761a922.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17575/HighFidelity-Beta-PR15009-761a922.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17575/HighFidelity-Beta-Interface-PR15009-761a922.exe

**Comment by hifi-gustavo** [comment]
Can someone from High Fidelity approve this PR?

**Comment by ingerjm0** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17877/HighFidelity-Beta-PR15009-761a922.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17877/HighFidelity-Quest-Beta-PR15009-761a922.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17877/HighFidelity-Beta-PR15009-761a922.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17877/HighFidelity-Beta-Interface-PR15009-761a922.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17877/HighFidelity-Beta-PR15009-761a922.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17877/HighFidelity-Beta-Interface-PR15009-761a922.exe

**Pull request merged**